assets/src/design-system/components/search/search.js

```diff
@@ -0,0 +1,340 @@
+ * This is to separate the inputValue from the selectedValue should the search demand selection from a list as opposed to be freeform.
+ * @param {boolean} props.hasError If true, input and hint (if present) will show error styles.
+ * @param {string} props.hint Hint text to display below input (optional). If not present, no hint text will display.
+ * @param {boolean} props.isRTL If true, arrow left will trigger down, arrow right will trigger up.
+ * @param {string} props.label If present, will display a text label above input. ariaLabel controls accessibility label of input since there are times search components are present without label text visible.
+ * @param {Object} props.menuStylesOverride should be formatted as a css template literal with styled components. Gives access to completely overriding menu styles (container div > ul > li).
+ * @param {Function} props.onMenuItemClick Triggered when a user clicks or presses 'Enter' on an option, or 'Enter' is pressed on input. Returns an object containing label and value keys.
+ * @param {Array} props.options All options, should contain either 1) objects with a label, value, anything else you need can be added and accessed through renderItem or 2) Objects containing a label and options, where options is structured as first option with array of objects containing at least value and label - this will create a nested list.
+ * @param {string} props.placeholder Placeholder text to display in input if no value is present.
+ * @param {string} props.placement placement passed to popover for where menu should expand, defaults to "bottom_end".
+ * @param {number} props.popupFillWidth Allows for an override of how much of popup width to take up for dropDown.
+ * @param {number} props.popupZIndex Allows for an override of the default popup z index (2).
+ * @param {Function} props.renderItem If present when menu is open, will override the base list items rendered for each option, the entire item and whether it is selected will be returned and allow you to style list items internal to a list item without affecting dropdown functionality.
+ * @param {string} props.selectedValue The selected value, should correspond to an object in the options array of objects.
+ *
+ */
+
+export const Search = ({
+  ariaInputLabel = DEFAULT_ARIA_INPUT_LABEL,
+  ariaClearLabel = DEFAULT_ARIA_CLEAR_LABEL,
+  disabled,
+  handleSearchValueChange,
+  hasError,
+  hint,
+  label,
+  onMenuItemClick,
+  options = [],
+  placeholder = DEFAULT_PLACEHOLDER,
+  placement = PLACEMENT.BOTTOM,
+  popupFillWidth = DEFAULT_POPUP_FILL_WIDTH,
+  popupZIndex,
+  selectedValue = {},
+  ...rest
+}) => {
+  const listId = useMemo(() => `list-${uuidv4()}`, []);
+  const inputId = useMemo(() => `select-button-${uuidv4()}`, []);
+  const clearId = useMemo(() => `clear-button-${uuidv4()}`, []);
+
+  const inputRef = useRef();
+  const menuRef = useRef();
+
+  const {
+    activeOption,
+    getActiveOption,
+    inputValue,
```

Comment:
Maybe this is just me, but when I read `inputValue` I think "This must be a string"! 

💭 Maybe we should rename this to `inputState` to imply that there's more than just a primitive value here?